pkg/dataset/dataset.go

```diff
@@ -0,0 +1,225 @@
+package dataset
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type Dataset interface {
+	GetID() string
+	Type() string
+	Length() int
+	Nth(i int) (string, error)
+	Range(i, j int) ([]string, error)
+}
+
+// ArrayDataset represents an array of generic JSON data.
+type ArrayDataset struct {
+	ID   string
+	Data []any
+}
+
+func (d *ArrayDataset) GetID() string {
+	return d.ID
+}
+
+func (d *ArrayDataset) Type() string {
+	return "array"
```

Comment:
Can the types for the datasets be constants somewhere?

Reply:
no longer relevant after the reimplementation